prawcore/auth.py

```diff
@@ -174,17 +176,17 @@ class DeviceIDAuthorizer(Authorizer):
     due to the lack of an associated reddit account.
     """
 
-    def __init__(self, authenticator, device_id):
+    def __init__(self, authenticator, device_id='DO_NOT_TRACK_THIS_DEVICE'):
         """Represents an app-only OAuth2 authorization for 'installed' apps.
 
         :param authenticator: An instance of :class:`Authenticator`.
-        :param device_id: A unique ID (20-30 character ASCII string)
+        :param device_id: (optional) A unique ID (20-30 character ASCII string)
             For more information about this parameter, see:
             https://github.com/reddit/reddit/wiki/OAuth2#application-only-oauth
         """
-        if authenticator.client_secret != '':
+        if authenticator.client_secret:
             raise InvalidInvocation(
-                'confindential client cannot instantiate this class')
+                'confidential client cannot instantiate this class')
```

Comment:
Now that I'm having a second look, perhaps it would make sense to say something like DeviceIDAuthorizor cannot be used with an Authenticator that has a client secret.

Similarly since client secret is optional, we should probably have a similar test added to the other authorizer class to ensure that ClientSecret exists.